diff --git a/Dockerfile b/Dockerfile
index 0831dd0..b9a5928 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM nimlang/nim:alpine as build
+FROM alpine as build
 
 ARG  REPO=https://github.com/goodtiding5/nitter.git
 
@@ -9,6 +9,10 @@ RUN apk update \
 	pcre \
 	unzip \
 	git \
+        gcc \
+        libc-dev \
+        nim \
+        nimble \
 &&  mkdir -p /build
 
 WORKDIR /build
@@ -16,7 +20,7 @@ WORKDIR /build
 RUN set -ex \
 &&  git clone $REPO . \
 &&  nimble install -y --depsOnly \
-&&  nimble build -y -d:release --passC:"-flto" --passL:"-flto" \
+&&  nimble build -y -d:release -d:zippyNoSimd --passC:"-flto" --passL:"-flto" \
 &&  strip -s nitter \
 &&  nimble scss \
 &&  nimble md
