From 2f9443e0185a24ca6b4103f97afff96cc5a25c73 Mon Sep 17 00:00:00 2001
From: Emilien Devos <contact@emiliendevos.be>
Date: Thu, 19 Jan 2023 09:08:51 +0100
Subject: [PATCH 1/1] use alpine edge for nim

---
 Dockerfile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index f65c3ed..ed40a2a 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,7 @@
-FROM nimlang/nim:1.6.10-alpine-regular as nim
+FROM alpine:latest as nim
 LABEL maintainer="setenforce@protonmail.com"
 
-RUN apk --no-cache add libsass-dev pcre
+RUN apk --no-cache add libsass-dev pcre gcc git libc-dev nim nimble
 
 WORKDIR /src/nitter
 
-- 
2.39.0
