From 1fcb5baac79870015de76f6802311efc2d87157d Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Sat, 21 Mar 2026 08:32:32 +0000
Subject: [PATCH 3/4] add x-cache header

---
 src/routes/rss.nim | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/routes/rss.nim b/src/routes/rss.nim
index 444dc70..8edd231 100644
--- a/src/routes/rss.nim
+++ b/src/routes/rss.nim
@@ -42,7 +42,7 @@ proc timelineRss*(req: Request; cfg: Config; query: Query; prefs: Prefs): Future
     let rss = renderTimelineRss(profile, cfg, prefs, multi=(names.len > 1))
     return Rss(feed: rss, cursor: profile.tweets.bottom)
 
-template respRss*(rss, page) =
+template respRss*(rss; page; cached = false) =
   if rss.cursor.len == 0:
     let info = case page
                of "User": " \"" & @"name" & "\" "
@@ -54,7 +54,7 @@ template respRss*(rss, page) =
     resp Http404, showError(getSuspended(@"name"), cfg)
 
   let headers = {"Content-Type": "application/rss+xml; charset=utf-8",
-                 "Min-Id": rss.cursor}
+                 "Min-Id": rss.cursor, "X-Cache": $cached}
   resp Http200, headers, rss.feed
 
 proc createRssRouter*(cfg: Config) =
@@ -77,7 +77,7 @@ proc createRssRouter*(cfg: Config) =
 
       var rss = await getCachedRss(key)
       if rss.cursor.len > 0:
-        respRss(rss, "Search")
+        respRss(rss, "Search", true)
 
       let tweets = await getGraphTweetSearch(query, cursor)
       rss.cursor = tweets.bottom
@@ -97,7 +97,7 @@ proc createRssRouter*(cfg: Config) =
 
       var rss = await getCachedRss(key)
       if rss.cursor.len > 0:
-        respRss(rss, "User")
+        respRss(rss, "User", true)
 
       rss = await timelineRss(request, cfg, Query(fromUser: @[name]), prefs)
 
@@ -132,7 +132,7 @@ proc createRssRouter*(cfg: Config) =
 
       var rss = await getCachedRss(key)
       if rss.cursor.len > 0:
-        respRss(rss, "User")
+        respRss(rss, "User", true)
 
       rss = await timelineRss(request, cfg, query, prefs)
 
@@ -168,7 +168,7 @@ proc createRssRouter*(cfg: Config) =
 
       var rss = await getCachedRss(key)
       if rss.cursor.len > 0:
-        respRss(rss, "List")
+        respRss(rss, "List", true)
 
       let
         list = await getCachedList(id=id)
-- 
2.53.0

