From 5266fddf2ee97169dbf391167dfd5164d4ea0709 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89milien=20Devos?= <contact@emiliendevos.be>
Date: Wed, 1 Jun 2022 20:38:01 +0000
Subject: [PATCH] Fix error when the ENV REDIS_PASS doesn't exist

---
 entrypoint.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/entrypoint.sh b/entrypoint.sh
index 1de8d79..25cc1c1 100755
--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -4,7 +4,7 @@ set -eu
 
 REDIS_HOST="${REDIS_HOST:-redis}"
 REDIS_PORT="${REDIS_PORT:-6379}"
-REDIS_PASS="${REDIS_PASS:-}"
+REDIS_PASS="${REDIS_PASS:-\"\"}"
 NITTER_HTTPS="${NITTER_HTTPS:-false}"
 NITTER_HOST="${NITTER_HOST:-nitter.net}"
 NITTER_NAME="${NITTER_NAME:-nitter}"
