--- Dockerfile.1	2022-05-20 16:04:10.693905933 +0000
+++ Dockerfile	2022-05-20 16:07:01.887924669 +0000
@@ -1,4 +1,4 @@
-FROM nimlang/nim:alpine as build
+FROM ghcr.io/maxisoft/nim-docker-images/nim as build
 
 ARG  REPO=https://github.com/zedeus/nitter.git
 
@@ -35,7 +35,7 @@
      NITTER_NAME="nitter" \
      NITTER_THEME="Nitter" \
      REPLACE_TWITTER="nitter.net" \
-     REPLACE_YOUTUBE="piped.kavin.rocks" \
+     REPLACE_YOUTUBE="invidio.us" \
      REPLACE_REDDIT="teddit.net" \
      REPLACE_INSTAGRAM=""
 
From 5266fddf2ee97169dbf391167dfd5164d4ea0709 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=89milien=20Devos?= <contact@emiliendevos.be>
Date: Wed, 1 Jun 2022 20:38:01 +0000
Subject: [PATCH] Fix error when the ENV REDIS_PASS doesn't exist

---
 entrypoint.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/entrypoint.sh b/entrypoint.sh
index 1de8d79..25cc1c1 100755
--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -4,7 +4,7 @@ set -eu
 
 REDIS_HOST="${REDIS_HOST:-redis}"
 REDIS_PORT="${REDIS_PORT:-6379}"
-REDIS_PASS="${REDIS_PASS:-}"
+REDIS_PASS="${REDIS_PASS:-\"\"}"
 NITTER_HTTPS="${NITTER_HTTPS:-false}"
 NITTER_HOST="${NITTER_HOST:-nitter.net}"
 NITTER_NAME="${NITTER_NAME:-nitter}"
